Review Documentation

malinibalachandran opened the pull request:
_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
_Provide a 1-2 line brief overview of the changes submitted through the Pull Request..._

### REFERENCES
Reference URL for issue tracking (Github): **\<URL to be filled>**

- [ ] Added label to the Pull Request for easier discoverability and search

### CODE MAINTAINABILITY
- [ ] **_Commit Message meets guidelines as indicated in the URL\*_**
	- https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [ ] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approve the Pull Request for merge are required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
	- [ ] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Adopted domain specific coding standards ([SBL Coding Standards](https://github.com/tianocore-docs/Docs/raw/master/Specifications/CCS_2_1_Draft.pdf))
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Tracing output are minimized and logic
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

# _Code must act as a teacher for future developers_

malinibalachandran committed:
Reviewed index.md

malinibalachandran committed:
Review platforms.md

malinibalachandran committed:
Review platforms.md

Changed title 
Added a reference to Hardware Setup

malinibalachandran committed:
Update platforms.md

malinibalachandran committed:
Reviewed camera_serial_number.md

Formatted and rephrased text for clarity and standards.

malinibalachandran committed:
Reviewed pipelinesetup.md

Rephrased text for clarity and formatted the content

malinibalachandran commented on docs_src/pipelinesetup.md:
```diff
@@ -1,54 +1,50 @@
 
-!!! note
-    To manually download models you can follow links provided in the [Model List](../configs/models/2022/models.list.yml)
+**_Note:_**  To manually download the models, see the links in [Model List](../configs/models/2022/models.list.yml).
```
The step instructs to run a command to manually download the models. Then, what does the second note mean? This is a bit confusing.

Brian-McGinn replied:
This should read "All models used can be found in the [Model List](../configs/models/2022/models.list.yml)"

malinibalachandran replied:
Are these the pre-trained models? Can it be "The pre-trained models are listed in Model List"?

Brian-McGinn replied:
Yes that would make sense

malinibalachandran commented on docs_src/pipelinesetup.md:
```diff
@@ -1,54 +1,50 @@
+**_Note:_**  To manually download the models, see the links in [Model List](../configs/models/2022/models.list.yml).
 
-## Step 4: Build the reference design Docker images
+### Step 4: Build the reference design Docker images
```
```suggestion
### Step 4: Build the reference design Docker* images
```

malinibalachandran committed:
Reviewed pipelinebenchmarking.md

malinibalachandran commented on docs_src/pipelinebenchmarking.md:
````diff
@@ -1,105 +1,102 @@
 make consolidate ROOT_DIRECTORY=<output dir>
 ```
-and this will give all the performance metrics among different workload cases given the same root directory specified by `ROOT_DIRECTORY` as shown above.
+This command will give all the performance metrics among different workload cases given the same root directory specified by `ROOT_DIRECTORY` as shown above.
````
This sentence is not clear. Does it mean that the command will give the performance metrics of all workload cases that are in the specified root directory?

Brian-McGinn replied:
What it is trying to say is that the consolidate command will consolidate (average) all of the metrics between benchmark runs that exist under the 'ROOT_DIRECTORY'. For example it will consolidate the FPS average between 'benchmark/5_object' and 'benchmark/10_object'.

malinibalachandran replied:
Rephrased. Please check

Brian-McGinn replied:
That looks good

malinibalachandran committed:
Reviewed faq.md

malinibalachandran committed:
Updated Heading

malinibalachandran commented on docs_src/faq.md:
```diff
@@ -1,25 +1,25 @@
 ## How do I simulate RTSP cameras?
 
-The camera simulator script is the best way to run simulated RTSP camera streams. Details about the script can be found on the [pipeline run](./pipelinerun.md#run-camera-simulator) page.
+You can use the camera simulator script to run simulated RTSP camera streams. For more details on the script, refer to [Run Pipeline](./pipelinerun.md#run-camera-simulator) page.
```
Do you want to add the camera simulator script name so that it is consistent with the other answers?

Brian-McGinn replied:
That is a great idea. `camera-simulator.sh` is the script name

malinibalachandran committed:
Reviewed hardwaresetup.md

malinibalachandran commented on docs_src/hardwaresetup.md:
```diff
@@ -1,52 +1,41 @@
-## 11th & 12th Gen Intel® Core™ Setup
+## Set up 11th Gen Intel® Core™ Processor and 12th Gen Intel® Core™ Processor
 
-### Step 1: Install Ubuntu 20.04
```
Please do check the version of Ubuntu. Is it different for different processors? If the version is the same, I suggest combining the steps for 11th and 12th Gen and Xeon as it is a repetition. You can add Xeon to the heading.

Brian-McGinn replied:
The versions are different for this release

malinibalachandran committed:
Reviewed v1-0-0.md

malinibalachandran committed:
Reviewed references.md

malinibalachandran committed:
Reviewed roadmap.md

malinibalachandran commented on docs_src/roadmap.md:
```diff
@@ -5,4 +5,89 @@
+We want to drive our roadmap by building the most valuable assets/ingredients to our partners. Our roadmap will be public and will provide complete transparency to ensure we receive continuous feedback from our partners. Raise any issues and requirements to help us guide our roadmap priorities, and in doing so we can drive the retail vertical forward. 
+
+Open an [issue on GitHub](https://github.com/intel-retail/automated-self-checkout/issues) to report a problem or to provide feedback.
+
```
Farhaan wanted the rows to be merged on the second column. It wasn't possible in Markdown. I have added a raw HTML table, which is being rendered in .md file. Please delete it if it doesn't make sense.

Brian-McGinn replied:
I was having a difficult time doing the same in Markdown so I think what you did was the better solution. The only addition I would suggest is to make both columns align to the top of the table.

malinibalachandran committed:
Reviewed pipelinerun.md

malinibalachandran commented on docs_src/pipelinerun.md:
```diff
@@ -1,128 +1,78 @@
-To disable classification process of image extraction when applying model, default is NOT disabling classification if this is not provided as input to docker-run.sh
+The following are the optional parameters that you can provide as input to `docker-run.sh`. Note that these parameters would affect the performance of the pipeline. 
+    
+- `--classification_disabled`: Disables the classification process of image extraction when applying model. By default, the classification is enabled. 
```
What does "when applying model" mean? It is not clear.

vli11 replied:
when taking the model IR files into the process to analyze the input source

malinibalachandran replied:
I am still did not understand the explanation. I have tried to rephrase:

Disables the classification process of image extraction when the model IR files are analyzed for the input source. By default, the classification is enabled. 


Is the second part of the sentence necessary. Can it be written as:

Disables the classification process of image extraction. By default, the classification is enabled.

Brian-McGinn replied:
I think the single sentence is clear enough.

Disables the classification process of image extraction. By default, the classification is enabled.

malinibalachandran commented on docs_src/pipelinerun.md:
```diff
@@ -1,128 +1,78 @@
 
-### `--color-width`
-Realsense camera color related property, to apply realsense camera color width, which will overwrite the default value of realsense gstreamer; if it's not provided, it will use the default value from realsense gstreamer; make sure to look up to your realsense camera's color capability using `rs-enumerate-devices`
+- `--color-height`: Realsense camera color related property, to apply realsense camera color height, which will overwrite the default value of realsense gstreamer; if it's not provided, it will use the default value from realsense gstreamer; make sure to look up to your realsense camera's color capability using `rs-enumerate-devices`
```
The description of all color parameters are not clear. Please check

Brian-McGinn replied:
RealSense cameras produce a color and depth frame. You can use the color-height, color-width, and color-framerate to customize the settings of the color frame output from the camera. There is a depth equivalent but we currently don't' support depth frames so they are not listed.

malinibalachandran replied:
Added this explanation for all three color parameters:

Allows you to customize the settings of the color frame output from the Intel® RealSense™ Cameras. This parameter will overwrite the default value of RealSense gstreamer. Use `rs-enumerate-devices` to look up the camera's color capability.

Brian-McGinn replied:
That looks much better!

malinibalachandran commented on docs_src/pipelinerun.md:
````diff
@@ -178,9 +122,68 @@ The output in results/pipeline0.log file lists FPS (frames per second) during pi
+```
+
+---
+## Run Camera Simulator
````
What is the need for running a camera simulator along with the pipeline? It would be good to add an explanation

vli11 replied:
In case users don't have camera device plugged into the system, they can run the camera simulator to view the pipeline analytic results based on a sample video file to mimic real time camera video.

Brian-McGinn replied:
Additionally the camera simulator can be used to infinitely loop through a video file for consistent benchmarking. For example if you want to validate that the performance is the same for 6 hours, 12 hours, and 24 hours. Looping the same video should produce the same results no matter how long you run.

malinibalachandran replied:
Added

malinibalachandran commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -176,10 +171,6 @@ sudo ./benchmark.sh --stream_density <target FPS> --logdir <output dir>/data --i
-Starts the camera simulator. To use, place the script in a folder named camera-simulator. At the same directory level as the camera-simulator folder, create a folder called sample-media. The camera-simulator.sh script will start a simulator for each .mp4 video that it finds in the sample-media folder and will enumerate them as camera_0, camera_1 etc.  Be sure the path to camera-simulator.sh scrip ...
-
-- **stop_server.sh**
+- `camera-simulator.sh`: This script starts the camera simulator. Create two folders named **camera-simulator** and **sample-media**. Place `camera-simulator.sh` in the **camera-simulator** folder. At the same directory level as the camera-simulator folder, create a folder called sample-media. The `camera-simulator.sh` script will start a simulator for each *.mp4* video that it finds in the **sam ...
```
The original text instructs to create a folder called sample-media, and that the script would start a simulator for each mp4 file in that folder. Does the script download the mp4 files automatically or should the user copy or download the files to the sample-media folder?

Brian-McGinn replied:
The camera-simulator will not automatically download media. The user will either need to copy their file into sample-media or run [download_sample_videos.sh](https://github.com/intel-retail/automated-self-checkout/blob/main/benchmark-scripts/download_sample_videos.sh)

malinibalachandran committed:
Reviewed query_usb_camera.md

malinibalachandran commented on docs_src/query_usb_camera.md:
```diff
@@ -1,25 +1,25 @@
-# How to query a usb camera
+# Query USB Camera
```
What is the need to query a USB camera? This page is not linked anywhere. How will the user reach this page?

Brian-McGinn replied:
Let me verify with the team but we may be able to remove this document.

vli11 replied:
somehow doc evolves and it was gone. I just added the comment on L37 of pipelinerun.md, that's where it should have link to query USB Camera

Brian-McGinn commented on docs_src/faq.md:
```diff
@@ -1,25 +1,25 @@
 ## What are the platform requirements?
 
-11th gen Intel processors or newer are supported but may not be optimal. For optimal setups see the [platform guide](./platforms.md)
+11th Gen Intel® Core™ Processors or newer processors are supported but may not be optimal. For optimal hardware, refer to the [platform guide](./platforms.md).
```
After reviewing this I think the first sentence can be removed and the answer will essentially be a link to the platform guide.

Brian-McGinn commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -1,105 +1,102 @@
-# Pipeline Benchmarking
+# Benchmark Pipeline
```
"Computer Vision Pipeline Benchmarking" might be a better title for the page

Brian-McGinn commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -1,105 +1,102 @@
 
-## Step 1: Build Benchmark Docker Images
-Benchmark scripts are containerized inside Docker, depending on platforms/hardware you have, refer to the following table to choose one to build:
+## Steps to Benchmark Pipeline
```
If the title changes we can follow this with 
## Steps to benchmark computer vision pipelines

Brian-McGinn commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -1,105 +1,102 @@
+## Steps to Benchmark Pipeline
+
+### Step 1: Build the benchmark Docker* images
+Benchmark scripts are containerized inside Docker. The following table lists the commands for various platforms. Choose and run the command corresponding to your platforms or hardware.
```
```suggestion
Benchmark scripts are containerized inside Docker. The following table lists the commands for various platforms. Choose and run the command corresponding to your hardware configuration.
```

Brian-McGinn commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -1,105 +1,102 @@
-## Step 2: Run Benchmark
-The `benchmark.sh` shell script is located under `benchmark_scripts` directory under the base directory.  Before executing this script,
-change the current directory to directory `benchmark_scripts`.
+| Intel platforms with Intel integrated GPUs | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-igt</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:igt`</b> images |
```
```suggestion
| Intel integrated GPUs | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-igt</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:igt`</b> images |
```

Brian-McGinn commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -1,105 +1,102 @@
-The `benchmark.sh` shell script is located under `benchmark_scripts` directory under the base directory.  Before executing this script,
-change the current directory to directory `benchmark_scripts`.
+| Intel platforms with Intel integrated GPUs | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-igt</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:igt`</b> images |
+| Intel platforms with Intel discrete GPUs   | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-xpu</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:xpu`</b> images |
```
```suggestion
| Intel discrete GPUs   | <pre>cd benchmark-scripts<br>make build-benchmark<br>make build-xpu</pre> | Docker images command to show both <b>`benchmark:dev`</b> and <b>`benchmark:xpu`</b> images |
```

Brian-McGinn commented on docs_src/pipelinebenchmarking.md:
```diff
@@ -117,22 +114,20 @@ One of the consolidation example outputs is shown below:
+### Benchmark Stream Density
 
-Another thing pipeline benchmarking can do is to discover the maximum number of workloads/streams that can be ran in parallel for a given target FPS.  This can be useful to determine the hardware requirements in order to achieve the desired performance for input sources.
+Benchmarking a pipeline can also discover the maximum number of workloads or streams that can be ran in parallel for a given target FPS. This information is useful to determine the hardware required to achieve the desired performance for input sources.
```
```suggestion
Benchmarking a pipeline can also discover the maximum number of workloads that can be ran in parallel for a given target FPS. This information is useful to determine the hardware required to achieve the desired performance for the workload.
```

Brian-McGinn commented on docs_src/pipelinesetup.md:
```diff
@@ -1,54 +1,50 @@
+## Next Steps
 
-Run a [use case/pipeline](./pipelinerun.md) or run a [benchmark for a use case/pipeline](./pipelinebenchmarking.md)
+Run a [use case or pipeline](./pipelinerun.md) or run a [benchmark for a use case or pipeline](./pipelinebenchmarking.md).
```
```suggestion
Run a [pipeline](./pipelinerun.md) or run a [benchmark for a pipeline](./pipelinebenchmarking.md).
```

Brian-McGinn commented:
If you can change the destination branch to "https://github.com/intel-retail/automated-self-checkout -> main" we will be able to directly merge from this PR

malinibalachandran committed:
Update pipelinebenchmarking.md

Rephrased: This command will consolidate the performance metrics that exist in the specified  'ROOT_DIRECTORY'.

malinibalachandran committed:
Update pipelinebenchmarking.md

Updated Appendix

malinibalachandran committed:
Update faq.md

Updated questions 1 and 4.

malinibalachandran committed:
Update docs_src/pipelinesetup.md

malinibalachandran committed:
Update pipelinesetup.md

Updated Note on Model List

malinibalachandran committed:
Update pipelinebenchmarking.md

Updated heading

malinibalachandran committed:
Update docs_src/pipelinesetup.md

Co-authored-by: brian-intel <brian.mcginn@intel.com>

malinibalachandran committed:
Update roadmap.md

Fixed alignment

malinibalachandran committed:
Update docs_src/pipelinebenchmarking.md

Co-authored-by: brian-intel <brian.mcginn@intel.com>

malinibalachandran committed:
Update docs_src/pipelinebenchmarking.md

Co-authored-by: brian-intel <brian.mcginn@intel.com>

malinibalachandran committed:
Update docs_src/pipelinebenchmarking.md

Co-authored-by: brian-intel <brian.mcginn@intel.com>

malinibalachandran committed:
Update pipelinebenchmarking.md

Updated Sub-heading

jim-wang-yutsung commented on docs_src/camera_serial_number.md:
````diff
@@ -1,24 +1,32 @@
-    Firmware Update Id            :     <not the serial number>
-```
\ No newline at end of file
+2. While in the container, run `rs-enumerate-devices` to list metadata of all attached Intel® RealSense™ products:
+
+    ```
+    /home/pipeline-zoo/workspace# rs-enumerate-devices
+    ```
+    
+    Expected result:
+    ```
+    Device info: 
+        Name                          :     Intel RealSense D435
+        Serial Number                 :----->serial number<------
+        Firmware Version              :     05.08.15.00
+        Recommended Firmware Version  :     05.13.00.50
+        Physical Port                 :     /sys/devices/pci0000:00/0000:00:14.0/usb2/2-3/2-3:1.0/video4linux/video0
+        Debug Op Code                 :     15
+        Advanced Mode                 :     YES
+        Product Id                    :     0B07
+        Camera Locked                 :     YES
+        Product Line                  :     D400
+        Asic Serial Number            :     <not the serial number>
+        Firmware Update Id            :     <not the serial number>
+    ```
````
should we also mention something like check if your realsense camera is plugin into your system if you don't see expected result?

vli11 commented on docs_src/pipelinerun.md:
```diff
@@ -1,128 +1,78 @@
-## Optional parameters
-The optional parameters you can apply to the docker-run.sh input, just note that they will affect the performance of pipeline run.
+**_Note:_**  The value of x in `dgpu.x` can be 0, 1, 2, and so on depending on the number of discrete GPUs in the system.
+    
```
**_Note:_** Here is [How_to_query_usb_camera](./query_usb_camera.md)

malinibalachandran replied:
What is the reason to query USB camera? Adding the explanation would make it clearer.

Brian-McGinn replied:
We would use the query steps to see the formats that the USB camera supports. We can word it like:

Follow these [steps](/How_to_query_usb_camera.md) to see the output formats supported by your USB camera.

malinibalachandran committed:
Update pipelinerun.md

Updated Run Camera Simulator Section

malinibalachandran committed:
Update pipelinerun.md

Added note related to USB camera

malinibalachandran committed:
Update pipelinerun.md

malinibalachandran committed:
Update camera_serial_number.md
